### DISCUSSION TITLE
Batch API requests?

### DISCUSSION THREAD

**Discussion by alexwennerberg** [Byron/google-apis-rs#274] [Q&A]
One of the things I use in the Python SDK all the time to improve performance is their wrapper around making batch requests: 
https://developers.google.com/drive/api/v3/batch
https://developers.google.com/api-client-library/python/guide/batch

I may be missing something, but I don't see any support for batch requests in this library. Is that on the roadmap?

Thanks!

**Comment by Byron** [comment]
This *should* be as easy as running `make update-json` on your machine. That will fetch all APIs discoverable by the google discovery API and update existing API descriptions, too.
That way you might see the respective endpoints show up.
I myself have not run this in a little while, but will happily try again once I have my new computer which provides much more power and thus developer happiness :D.

**Comment by alexwennerberg** [comment]
Unfortunately the batch method isn't discoverable by the Google API, it has a different syntax. It would involve creating request JSONs and then merging them into a single multipart upload, as shown here: https://developers.google.com/drive/api/v3/batch#example

Thanks!

**Comment by Byron** [comment]
Oh, I see! Thanks for bringing this to my attention. If (or when) the next version of this generator is created to enable proper async-everything, it should be made to allow custom implementations to be added, too, just for these cases.
In the mean-time, I don't think I will touch this code-base except for the occasional `make update-json` and republish.
